src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c

```diff
@@ -73,5 +73,251 @@ void Reset_thrusterPlatformReference(thrusterPlatformReferenceConfig *configData
+        double cosGamma2 = (a*a + b*b - c2*c2) / (2*a*b);
+
+        psi = asin( fmin( fmax( (c1*sin(nu)*cosGamma2 - c2*sin(beta)*cosGamma1)/b, -1 ), 1 ) );
+    }
+    
+    double e_psi[3];
+    v3Cross(THat_F, r_CT_F, e_psi);
+    v3Normalize(e_psi, e_psi);
+
+    // compute intermediate rotation F2F1
+    double PRV_psi[3];
+    v3Scale(psi, e_psi, PRV_psi);
+    PRV2C(PRV_psi, F2F1);
+}
+
+void computeThirdRotation(double e_theta[3], double F2M[3][3], double F3F2[3][3])
+{
+    double e1 = e_theta[0];  
+    double e2 = e_theta[1];  
+    double e3 = e_theta[2];
+
+    double A = 2 * (F2M[1][0]*e2*e2 + F2M[0][0]*e1*e2 + F2M[2][0]*e2*e3) - F2M[1][0];
+    // double A = 2 * (F2M[0][0]*e1*e2 + F2M[2][0]*e2*e3) + F2M[1][0]*(e2*e2 - e1*e1 - e3*e3);
+    double B = 2 * (F2M[2][0]*e1 - F2M[0][0]*e3);
+    double C = F2M[1][0];
+    double Delta = B*B - 4*A*C;
+
+    /* compute exact solution or best solution depending on Delta */
+    double t, t1, t2, y, y1, y2, theta;
+    if (fabs(A) < PLATFORM_REFERENCE_EPS) {
+        if (fabs(B) < PLATFORM_REFERENCE_EPS) {
+            // zero-th order equation has no solution 
+            // the solution of the minimum problem is theta = MPI
+            theta = MPI;
+        }
+        else {
+            // first order equation
+            t = - C / B;
+            theta = 2*atan(t);
+        }
+    }
+    else {
+        if (Delta < 0) {
+            // second order equation has no solution 
+            // the solution of the minimum problem is found
+            if (fabs(B) < PLATFORM_REFERENCE_EPS) {
+                t = 0.0;
+            }
+            else {
+                double q = (A-C) / B;
+                t1 = (q + sqrt(q*q + 1));
+                t2 = (q - sqrt(q*q + 1));
+                y1 = (A*t1*t1 + B*t1 + C) / (1 + t1*t1);
+                y2 = (A*t2*t2 + B*t2 + C) / (1 + t2*t2);
+
+                // choose which returns a smaller fcn value between t1 and t2
+                t = t1;
+                if (fabs(y2) < fabs(y1)) {
+                    t = t2;
+                }
+            }
+            theta = 2*atan(t);
+            y = (A*t*t + B*t + C) / (1 + t*t);
+            // check if the absolute fcn minimum is for theta = MPI
+            if (fabs(A) < fabs(y)) {
+                theta = MPI;
+            }
+        }
+        else {
+            t1 = (-B + sqrt(Delta)) / (2*A);
+            t2 = (-B - sqrt(Delta)) / (2*A);
+            t = t1;
+            if (fabs(t2) < fabs(t1)) {
+                t = t2;
+            }
+
+            theta = 2*atan(t);
+        }
+    }
+
+    // compute intermediate rotation F3F2
+    double PRV_theta[3];
+    v3Scale(theta, e_theta, PRV_theta);
+    PRV2C(PRV_theta, F3F2);
+}
+
+void computeFinalRotation(double r_CM_M[3], double r_TM_F[3], double T_F[3], double FM[3][3])
+{
+    /*! define unit vectors of CM direction in M coordinates and thrust direction in F coordinates */
+    double rHat_CM_M[3];    v3Normalize(r_CM_M, rHat_CM_M);
```

Comment:
:small_red_triangle_down: Each statement should be on a new line. Let's do that through out.

Reply:
Fixed it.